src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -93,17 +108,18 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
       p => if (p.externalId.isEmpty) { false } else { existingExternalIds.contains(p.externalId.get) }
     )
 
-    val idMap = client.assets
-      .retrieveByExternalIds(existingExternalIds)
-      .map(_.map(e => e.externalId -> e.id).toMap)
-
     val create =
       if (assetsToCreate.isEmpty) { IO.unit } else client.assets.create(assetsToCreate)
     val update =
       if (assetsToUpdate.isEmpty) { IO.unit } else {
-        idMap.flatMap(idMap =>
-          client.assets.update(assetsToUpdate.map(a =>
-            a.into[AssetUpdate].withFieldComputed(_.id, x => idMap(x.externalId)).transform)))
+        client.assets.updateByExternalId(
+          assetsToUpdate
+            .map(
+              a =>
+                (
+                  a.externalId.getOrElse(""),
```

Comment:
Same thing here, those should always have an `.externalId` if we do `.partition` correctly, right? So better to use `.get` and crash if they don't.